neps/nep-0509.md

```diff
@@ -0,0 +1,231 @@
+
+## Reference Implementation
+
+TODO: This is essentially going to be describing the exact structure of `ChunkStateWitness`, `ChunkEndorsement`, and describing the exact algorithm to be used for the chunk validator shuffling.
+
+[This technical section is required for Protocol proposals but optional for other categories. A draft implementation should demonstrate a minimal implementation that assists in understanding or implementing this proposal. Explain the design in sufficient detail that:
+
+* Its interaction with other features is clear.
+* Where possible, include a Minimum Viable Interface subsection expressing the required behavior and types in a target programming language. (ie. traits and structs for rust, interfaces and classes for javascript, function signatures and structs for c, etc.)
+* It is reasonably clear how the feature would be implemented.
+* Corner cases are dissected by example.
+* For protocol changes: A link to a draft PR on nearcore that shows how it can be integrated in the current code. It should at least solve the key technical challenges.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.]
+
+## Validator Role Change
+Currently, there are two different types of validators and their responsibilities are as follows:
+|  | Top ~50% validators | Remaining validatiors (Chunk only producers) |
+|-----|:-----:|:----:|
+| block production | Y | N |
+| chunk production | Y | Y |
+| block validation | Y | N |
+
+With stateless validation, this structure does not make sense anymore for several reasons:
+* Chunk production is the most resource consuming activity.
+* (Only) chunk production needs state in memory while other responsibilities can be completed via acquiring state witness
+* Chunk production does not have to be performed by all validators. 
+
+Hence, the most simple proposal is to change Chunk-only producers to Chunk-only validators as follows:
+| | Top ~50% validators | Remaining validatiors (Chunk-only validators) |
+|-----|:-----:|:----:|
+| block production | Y | N |
+| chunk production | Y | N |
+| block validation | Y | N |
+| chunk validation | Y | Y |
+
+Block production and validation remain as responsibility of validators with more stake to maintain the same level of security.
+
+This approach is the most straight forward as it maintains the same grouping as we have today.
+
+Potential improvement to which can lower hardware requirement for more validators is limiting the responsibility of chunk production to top N validators, who are often equipped with powerful machines already.
+|  | Top N validatiors (Chunk proposers) | Top ~50% - N validators | Remaining validators (Chunk-only validators) |
+|-----|:-----:|:----:|:----:|
+| block production | Y | Y | N |
+| chunk production | Y | N | N |
+| block validation | Y | Y | N |
+| chunk validation | Y | Y | N |
+
+## Security Implications
```

Comment:
cc. @jancionear @pugachAG

Reply:
added in dd9ba0901d72473d17b0897170f4789bf8d4ffae